### ISSUE TITLE
Stage 4 with igw role fails to start lrbd.service

### ISSUE THREAD

**Issue by smithfarm** [SUSE/DeepSea#1385]
### Description of Issue/Question

A new failure appeared recently in our `deepsea/tier2/openattic` test case: Stage 4 fails to start the lrbd.service:

```
2018-09-09T09:20:59.816 INFO:teuthology.orchestra.run.target147135133161.stderr:+ tee /tmp/deepsea.4.log
2018-09-09T09:21:00.244 INFO:teuthology.orchestra.run.target147135133161.stdout:Starting stage: ceph.stage.4
2018-09-09T09:23:05.265 INFO:teuthology.orchestra.run.target147135133161.stdout:Parsing stage ceph.stage.4 steps... done
2018-09-09T09:23:05.266 INFO:teuthology.orchestra.run.target147135133161.stdout:
2018-09-09T09:23:05.266 INFO:teuthology.orchestra.run.target147135133161.stdout:Stage initialization output:
2018-09-09T09:23:05.267 INFO:teuthology.orchestra.run.target147135133161.stdout:
2018-09-09T09:23:05.267 INFO:teuthology.orchestra.run.target147135133161.stdout:
2018-09-09T09:23:05.267 INFO:teuthology.orchestra.run.target147135133161.stdout:[init] Executing runner validate.deploy... ok
2018-09-09T09:23:05.267 INFO:teuthology.orchestra.run.target147135133161.stdout:[init] Executing runner select.minions... ok
2018-09-09T09:23:05.268 INFO:teuthology.orchestra.run.target147135133161.stdout:[init] Executing runner select.one_minion... ok
2018-09-09T09:23:05.268 INFO:teuthology.orchestra.run.target147135133161.stdout:[init] Executing runner select.minions... ok
2018-09-09T09:23:05.268 INFO:teuthology.orchestra.run.target147135133161.stdout:[init] Executing runner cephprocesses.need_restart_config_change...
2018-09-09T09:23:05.268 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  cephprocesses.need_restart_config_change(role=mds)...
2018-09-09T09:23:05.268 INFO:teuthology.orchestra.run.target147135133161.stdout:               in target147135133159.teuthology... ok
2018-09-09T09:23:05.268 INFO:teuthology.orchestra.run.target147135133161.stdout:               in target147135133137.teuthology... ok
2018-09-09T09:23:05.269 INFO:teuthology.orchestra.run.target147135133161.stdout:[init] Executing runner cephprocesses.need_restart_config_change... ok
2018-09-09T09:23:05.269 INFO:teuthology.orchestra.run.target147135133161.stdout:[init] Executing runner cephprocesses.need_restart_lsof...
2018-09-09T09:23:05.269 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  cephprocesses.need_restart_lsof(role=mds)...
2018-09-09T09:23:05.269 INFO:teuthology.orchestra.run.target147135133161.stdout:               in target147135133159.teuthology... ok
2018-09-09T09:23:05.269 INFO:teuthology.orchestra.run.target147135133161.stdout:               in target147135133137.teuthology... ok
2018-09-09T09:23:05.269 INFO:teuthology.orchestra.run.target147135133161.stdout:[init] Executing runner cephprocesses.need_restart_lsof... ok
2018-09-09T09:23:05.270 INFO:teuthology.orchestra.run.target147135133161.stdout:[init] Executing runner select.minions... ok
2018-09-09T09:23:05.270 INFO:teuthology.orchestra.run.target147135133161.stdout:[init] Executing runner select.minions... ok
2018-09-09T09:23:05.270 INFO:teuthology.orchestra.run.target147135133161.stdout:[init] Executing runner select.minions... ok
2018-09-09T09:23:05.270 INFO:teuthology.orchestra.run.target147135133161.stdout:[init] Executing runner sharedsecret.show... ok
2018-09-09T09:23:05.270 INFO:teuthology.orchestra.run.target147135133161.stdout:
2018-09-09T09:23:05.270 INFO:teuthology.orchestra.run.target147135133161.stdout:[1/34] Executing state mine.send...
2018-09-09T09:23:05.270 INFO:teuthology.orchestra.run.target147135133161.stdout:         target147135133161.teuthology... ok
2018-09-09T09:23:05.271 INFO:teuthology.orchestra.run.target147135133161.stdout:[2/34] Executing state ceph.igw.config...
2018-09-09T09:23:05.271 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  select.minions(cluster=ceph, roles=igw)... ok
2018-09-09T09:23:05.271 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  target147135133161.teuthology: demo pool(ceph osd pool create iscsi-images 128) ok
2018-09-09T09:23:05.271 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  target147135133161.teuthology: demo pool enable application(ceph osd pool application enable iscsi-images rbd || :) ok
2018-09-09T09:23:05.271 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  target147135133161.teuthology: demo image(rbd -p iscsi-images create demo --size=1024) ok
2018-09-09T09:23:05.271 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  select.minions(cluster=ceph, roles=igw)... ok
2018-09-09T09:23:05.272 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  select.public_addresses(cluster=ceph, tuples=True, roles=igw)...
2018-09-09T09:23:05.272 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  public.address()...
2018-09-09T09:23:05.272 INFO:teuthology.orchestra.run.target147135133161.stdout:               in target147135133159.teuthology... ok
2018-09-09T09:23:05.272 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  select.public_addresses(cluster=ceph, tuples=True, roles=igw)... ok
2018-09-09T09:23:05.272 INFO:teuthology.orchestra.run.target147135133161.stdout:         target147135133161.teuthology... ok
2018-09-09T09:23:05.272 INFO:teuthology.orchestra.run.target147135133161.stdout:[3/34] Executing state ceph.igw.auth...
2018-09-09T09:23:05.273 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  select.minions(cluster=ceph, host=True, roles=igw)... ok
2018-09-09T09:23:05.273 INFO:teuthology.orchestra.run.target147135133161.stdout:         target147135133161.teuthology... ok
2018-09-09T09:23:05.273 INFO:teuthology.orchestra.run.target147135133161.stdout:[4/34] Executing state ceph.igw.keyring...
2018-09-09T09:23:05.273 INFO:teuthology.orchestra.run.target147135133161.stdout:         target147135133159.teuthology... ok
2018-09-09T09:23:05.274 INFO:teuthology.orchestra.run.target147135133161.stdout:[5/34] Executing state ceph.igw.sysconfig...
2018-09-09T09:23:05.274 INFO:teuthology.orchestra.run.target147135133161.stdout:         target147135133159.teuthology... ok
2018-09-09T09:23:05.274 INFO:teuthology.orchestra.run.target147135133161.stdout:[6/34] Executing state ceph.igw.import...
2018-09-09T09:23:05.274 INFO:teuthology.orchestra.run.target147135133161.stdout:         target147135133159.teuthology... ok
2018-09-09T09:23:05.274 INFO:teuthology.orchestra.run.target147135133161.stdout:[7/34] Executing state ceph.igw...
2018-09-09T09:23:05.274 INFO:teuthology.orchestra.run.target147135133161.stdout:         target147135133159.teuthology... fail
2018-09-09T09:23:05.274 INFO:teuthology.orchestra.run.target147135133161.stdout:[8/34] Executing state ceph.wait...
2018-09-09T09:23:05.275 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  target147135133161.teuthology: wait(wait.out) ok
2018-09-09T09:23:05.275 INFO:teuthology.orchestra.run.target147135133161.stdout:         target147135133161.teuthology... ok
2018-09-09T09:23:05.275 INFO:teuthology.orchestra.run.target147135133161.stdout:[9/34] Executing state ceph.processes.igw...
2018-09-09T09:23:05.275 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  target147135133159.teuthology: wait for igw processes(cephprocesses.wait) ok
2018-09-09T09:23:05.275 INFO:teuthology.orchestra.run.target147135133161.stdout:         target147135133159.teuthology... ok
2018-09-09T09:23:05.275 INFO:teuthology.orchestra.run.target147135133161.stdout:[10/34] Executing state ceph.igw.restart...
2018-09-09T09:23:05.275 INFO:teuthology.orchestra.run.target147135133161.stdout:         |_  target147135133159.teuthology: restart(systemctl reload lrbd) fail
2018-09-09T09:23:05.276 INFO:teuthology.orchestra.run.target147135133161.stdout:         target147135133159.teuthology... fail
2018-09-09T09:23:05.276 INFO:teuthology.orchestra.run.target147135133161.stdout:Finished stage ceph.stage.4: succeeded=8/34 failed=2/34
2018-09-09T09:23:05.276 INFO:teuthology.orchestra.run.target147135133161.stdout:
2018-09-09T09:23:05.276 INFO:teuthology.orchestra.run.target147135133161.stdout:Failures summary:
2018-09-09T09:23:05.276 INFO:teuthology.orchestra.run.target147135133161.stdout:ceph.igw (/srv/salt/ceph/igw):
2018-09-09T09:23:05.276 INFO:teuthology.orchestra.run.target147135133161.stdout:  target147135133159.teuthology:
2018-09-09T09:23:05.277 INFO:teuthology.orchestra.run.target147135133161.stdout:    enable lrbd: Service lrbd has been enabled, and is dead
2018-09-09T09:23:05.277 INFO:teuthology.orchestra.run.target147135133161.stdout:ceph.igw.restart (/srv/salt/ceph/igw/restart):
2018-09-09T09:23:05.277 INFO:teuthology.orchestra.run.target147135133161.stdout:  target147135133159.teuthology:
2018-09-09T09:23:05.277 INFO:teuthology.orchestra.run.target147135133161.stdout:    restart: Command "systemctl reload lrbd" run
2018-09-09T09:23:05.277 INFO:teuthology.orchestra.run.target147135133161.stdout:        stdout:
2018-09-09T09:23:05.277 INFO:teuthology.orchestra.run.target147135133161.stdout:        stderr: Job for lrbd.service invalid.
```

`systemctl status lrbd.service` shows:

```
(to be provided)
```

What other debugging info is needed?

**Comment by smithfarm** [comment]
Logs of some jobs that failed due to this issue:

* http://149.202.174.223/ubuntu-2018-09-09_09:00:31-suse:tier2:openattic-ses5---basic-openstack/168/teuthology.log
* http://149.202.174.223/ubuntu-2018-09-08_19:18:55-suse-ses5---basic-openstack/124/teuthology.log
* http://149.202.174.223/ubuntu-2018-09-08_19:18:55-suse-ses5---basic-openstack/131/teuthology.log

**Comment by smithfarm** [comment]
Happening in the Staging CI with deepsea 0.8.6:

* https://ci.suse.de/job/storage-qa-teuthology-smoke-devel-5.0/lastCompletedBuild/testReport/teuthology/Devel/suse_tier2_igw__clusters_2nodes4disks_yaml_distros_sle_12_3_yaml_exec_health_igw_yaml_tasks_yaml_/attachments/teuthology-suse-382.log

**Comment by smithfarm** [comment]
In all likelihood, this was due to an outdated SLE-12-SP3 OpenStack image.

https://ci.suse.de/job/storage-qa-teuthology-smoke-devel-5.0/lastCompletedBuild/testReport/ is now back in the green!

**Comment by smithfarm** [comment]
This bug is still live in the CI. See latest Jenkins build from today: https://ci.suse.de/job/storage-run-teuthology-5.0/lastCompletedBuild/testReport/teuthology/Devel/suse_tier2_igw__clusters_2nodes4disks_yaml_distros_sle_12_3_yaml_exec_health_igw_cli_yaml_tasks_yaml_/attachments/teuthology-suse-3848.log

**Comment by smithfarm** [comment]
Potential workaround from @swiftgist :

In lrbd, change line 1270 from

```
        for cmd in uniq(self.cmds):
            popen(cmd)
        self._enable_rbd()
```

to

```
        for cmd in uniq(self.cmds):
            retry(cmd)
        self._enable_rbd()
```

**Comment by jschmid1** [comment]
Can we have a patch for lrbd then? @smithfarm @swiftgist ?

**Comment by smithfarm** [comment]
@jschmid1 As of yesterday, the patched lrbd is already in Devel:Storage:5.0 and has been submitted to maintenance.

It's still too early to tell if this issue can be closed, though.

**Comment by jschmid1** [comment]
thanks @smithfarm. Let's monitor that

**Comment by denisok** [comment]
```
2018-09-28T07:28:30.186603+00:00 target192168000071 systemd[1]: Started /usr/bin/systemctl start lrbd.service.
2018-09-28T07:28:30.190633+00:00 target192168000071 systemd[1]: Starting configures target.service from Ceph...
2018-09-28T07:28:32.161941+00:00 target192168000071 systemd[1]: Mounting Configuration File System...
2018-09-28T07:28:32.168344+00:00 target192168000071 systemd[1]: Mounted Configuration File System.
2018-09-28T07:28:32.192512+00:00 target192168000071 lrbd[26827]: ERROR: command failed
2018-09-28T07:28:32.192953+00:00 target192168000071 lrbd[26827]: rbd -p iscsi-images --name client.igw.target192168000071 map demo
2018-09-28T07:28:32.193353+00:00 target192168000071 lrbd[26827]: /dev/rbd0
2018-09-28T07:28:32.193540+00:00 target192168000071 lrbd[26827]: targetcli /backstores/rbd create name=iscsi-images-demo dev=/dev/rbd/iscsi-images/demo
2018-09-28T07:28:32.193766+00:00 target192168000071 lrbd[26827]: Warning: Could not load preferences file /root/.targetcli/prefs.bin.
2018-09-28T07:28:32.193947+00:00 target192168000071 lrbd[26827]: mount: none is already mounted or /sys/kernel/config busy
2018-09-28T07:28:32.194128+00:00 target192168000071 lrbd[26827]: Traceback (most recent call last):
2018-09-28T07:28:32.194331+00:00 target192168000071 lrbd[26827]:   File "/usr/bin/targetcli", line 82, in <module>
2018-09-28T07:28:32.194513+00:00 target192168000071 lrbd[26827]:     main()
2018-09-28T07:28:32.194691+00:00 target192168000071 lrbd[26827]:   File "/usr/bin/targetcli", line 71, in main
2018-09-28T07:28:32.194874+00:00 target192168000071 lrbd[26827]:     shell.run_cmdline(" ".join(sys.argv[1:]))
2018-09-28T07:28:32.195060+00:00 target192168000071 lrbd[26827]:   File "/usr/lib/python2.7/site-packages/configshell/shell.py", line 934, in run_cmdline
2018-09-28T07:28:32.195243+00:00 target192168000071 lrbd[26827]:     self._execute_command(path, command, pparams, kparams)
2018-09-28T07:28:32.195419+00:00 target192168000071 lrbd[26827]:   File "/usr/lib/python2.7/site-packages/configshell/shell.py", line 899, in _execute_command
2018-09-28T07:28:32.195625+00:00 target192168000071 lrbd[26827]:     self.log.error(msg)
2018-09-28T07:28:32.195802+00:00 target192168000071 lrbd[26827]:   File "/usr/lib/python2.7/site-packages/configshell/log.py", line 159, in error
2018-09-28T07:28:32.195979+00:00 target192168000071 lrbd[26827]:     self._log('error', msg)
2018-09-28T07:28:32.196170+00:00 target192168000071 lrbd[26827]:   File "/usr/lib/python2.7/site-packages/configshell/log.py", line 112, in _log
2018-09-28T07:28:32.196351+00:00 target192168000071 lrbd[26827]:     self.con.display(msg)
2018-09-28T07:28:32.196524+00:00 target192168000071 lrbd[26827]:   File "/usr/lib/python2.7/site-packages/configshell/console.py", line 153, in display
2018-09-28T07:28:32.196720+00:00 target192168000071 lrbd[26827]:     self.raw_write(text)
2018-09-28T07:28:32.196911+00:00 target192168000071 lrbd[26827]:   File "/usr/lib/python2.7/site-packages/configshell/console.py", line 141, in raw_write
2018-09-28T07:28:32.197217+00:00 target192168000071 lrbd[26827]:     self._stdout.write(text)
2018-09-28T07:28:32.197406+00:00 target192168000071 lrbd[26827]: TypeError: expected a string or other character buffer object
2018-09-28T07:28:32.201733+00:00 target192168000071 systemd[1]: lrbd.service: Main process exited, code=exited, status=1/FAILURE
2018-09-28T07:28:32.202272+00:00 target192168000071 systemd[1]: Failed to start configures target.service from Ceph.
2018-09-28T07:28:32.202505+00:00 target192168000071 systemd[1]: lrbd.service: Unit entered failed state.
2018-09-28T07:28:32.202704+00:00 target192168000071 systemd[1]: lrbd.service: Failed with result 'exit-code'.
2018-09-28T07:28:32.203886+00:00 target192168000071 salt-minion[3611]: [ERROR   ] Command '['systemd-run', '--scope', 'systemctl', 'start', 'lrbd.service']' failed with return code: 1
2018-09-28T07:28:32.204235+00:00 target192168000071 salt-minion[3611]: [ERROR   ] output: Running scope as unit run-r0d74efe41e9d484dbb2f5f060f417462.scope.
2018-09-28T07:28:32.204433+00:00 target192168000071 salt-minion[3611]: Job for lrbd.service failed because the control process exited with error code. See "systemctl status lrbd.service" and "journalctl -xe" for details.
2018-09-28T07:28:32.205885+00:00 target192168000071 salt-minion[3611]: [INFO    ] Executing command ['systemctl', 'is-enabled', 'lrbd.service'] in directory '/root'
2018-09-28T07:28:32.224120+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] output: disabled
2018-09-28T07:28:32.225715+00:00 target192168000071 salt-minion[3611]: [INFO    ] Executing command ['systemctl', 'is-enabled', 'lrbd.service'] in directory '/root'
2018-09-28T07:28:32.242628+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] output: disabled
2018-09-28T07:28:32.242941+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] Service 'lrbd' is not masked
2018-09-28T07:28:32.244253+00:00 target192168000071 salt-minion[3611]: [INFO    ] Executing command ['systemd-run', '--scope', 'systemctl', 'enable', 'lrbd.service'] in directory '/root'
2018-09-28T07:28:32.275014+00:00 target192168000071 systemd[1]: Started /usr/bin/systemctl enable lrbd.service.
2018-09-28T07:28:32.277519+00:00 target192168000071 systemd[1]: Reloading.
2018-09-28T07:28:32.351538+00:00 target192168000071 systemd[1]: nss-lookup.target: Dependency Before=nss-lookup.target dropped
2018-09-28T07:28:32.376388+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] output: Running scope as unit run-re4421123b3a545dd8daa03fc6a8881e4.scope.
2018-09-28T07:28:32.376631+00:00 target192168000071 salt-minion[3611]: Created symlink from /etc/systemd/system/multi-user.target.wants/lrbd.service to /usr/lib/systemd/system/lrbd.service.
2018-09-28T07:28:32.377479+00:00 target192168000071 salt-minion[3611]: [INFO    ] Executing command ['systemctl', 'is-enabled', 'lrbd.service'] in directory '/root'
2018-09-28T07:28:32.396236+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] output: enabled
2018-09-28T07:28:32.396697+00:00 target192168000071 salt-minion[3611]: [ERROR   ] {'lrbd': True}
```

**Comment by denisok** [comment]
```
2018-09-28T07:28:27.804765+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] LazyLoaded cmd.run
2018-09-28T07:28:27.805591+00:00 target192168000071 salt-minion[3611]: [INFO    ] Running state [. /etc/sysconfig/lrbd; lrbd -v $LRBD_OPTIONS -f /tmp/lrbd.conf] at time 07:28:27.805328
2018-09-28T07:28:27.805837+00:00 target192168000071 salt-minion[3611]: [INFO    ] Executing state cmd.run for . /etc/sysconfig/lrbd; lrbd -v $LRBD_OPTIONS -f /tmp/lrbd.conf
2018-09-28T07:28:27.808146+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] LazyLoaded cmd.retcode
2018-09-28T07:28:27.808524+00:00 target192168000071 salt-minion[3611]: [INFO    ] Executing command 'cat /tmp/lrbd.conf | grep -q "\"pools\": \[\]$"' in directory '/root'
2018-09-28T07:28:27.824925+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] output:
2018-09-28T07:28:27.825549+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] Last command return code: 1
2018-09-28T07:28:27.826189+00:00 target192168000071 salt-minion[3611]: [INFO    ] Executing command '. /etc/sysconfig/lrbd; lrbd -v $LRBD_OPTIONS -f /tmp/lrbd.conf' in directory '/root'
2018-09-28T07:28:27.829377+00:00 target192168000071 salt-minion[3611]: [WARNING ] Unable to find IPv6 record for "target192168000071.teuthology" causing a 10 second timeout when rendering grains. Set the dns or /etc/hosts for IPv6 to clear this.
2018-09-28T07:28:27.987984+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] Please install 'virt-what' to improve results of the 'virtual' grain.
2018-09-28T07:28:28.045927+00:00 target192168000071 salt-minion[3611]: [ERROR   ] Exception during resolving address: [Errno 1] Unknown host
2018-09-28T07:28:28.165696+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] stderr: INFO: Executing /usr/sbin/lrbd -v -n client.igw.target192168000071 -f /tmp/lrbd.conf -n client.igw.target192168000071
2018-09-28T07:28:28.165999+00:00 target192168000071 salt-minion[3611]: INFO: No object lrbd.conf to remove from pool write_test
2018-09-28T07:28:28.166186+00:00 target192168000071 salt-minion[3611]: INFO: No object lrbd.conf to remove from pool cephfs_data
2018-09-28T07:28:28.166398+00:00 target192168000071 salt-minion[3611]: INFO: No object lrbd.conf to remove from pool cephfs_metadata
2018-09-28T07:28:28.166603+00:00 target192168000071 salt-minion[3611]: INFO: No object lrbd.conf to remove from pool iscsi-images
2018-09-28T07:28:28.166806+00:00 target192168000071 salt-minion[3611]: INFO: No object lrbd.conf to remove from pool .rgw.root
2018-09-28T07:28:28.166991+00:00 target192168000071 salt-minion[3611]: INFO: No object lrbd.conf to remove from pool default.rgw.control
2018-09-28T07:28:28.167195+00:00 target192168000071 salt-minion[3611]: INFO: No object lrbd.conf to remove from pool default.rgw.meta
2018-09-28T07:28:28.167382+00:00 target192168000071 salt-minion[3611]: INFO: No object lrbd.conf to remove from pool default.rgw.log
2018-09-28T07:28:28.167576+00:00 target192168000071 salt-minion[3611]: [INFO    ] {'pid': 26763, 'retcode': 0, 'stderr': 'INFO: Executing /usr/sbin/lrbd -v -n client.igw.target192168000071 -f /tmp/lrbd.conf -n client.igw.target192168000071\nINFO: No object lrbd.conf to remove from pool write_test\nINFO: No object lrbd.conf to remove from pool cephfs_data\nINFO: No object lrbd.conf to remove from pool cephfs_metadata\nINFO: No object lrbd.conf to remove from pool iscsi-images\nINFO: No object lrbd.conf to remove from pool .rgw.root\nINFO: No object lrbd.conf to remove from pool default.rgw.control\nINFO: No object lrbd.conf to remove from pool default.rgw.meta\nINFO: No object lrbd.conf to remove from pool default.rgw.log', 'stdout': ''}
2018-09-28T07:28:28.169442+00:00 target192168000071 salt-minion[3611]: [INFO    ] Completed state [. /etc/sysconfig/lrbd; lrbd -v $LRBD_OPTIONS -f /tmp/lrbd.conf] at time 07:28:28.166315 duration_in_ms=360.987
2018-09-28T07:28:28.169683+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] File /var/cache/salt/minion/accumulator/140278201325648 does not exist, no need to cleanup.
2018-09-28T07:28:28.177294+00:00 target192168000071 salt-minion[3611]: [DEBUG   ] Minion return retry timer set to 10 seconds (randomized)
2018-09-28T07:28:28.177851+00:00 target192168000071 salt-minion[3611]: [INFO    ] Returning information for job: 20180928072820907953
```

**Comment by denisok** [comment]
nothing special in kernel log:
```
2018-09-28T07:28:31.893721+00:00 target192168000071 kernel: [ 1522.414174] Key type ceph registered
2018-09-28T07:28:31.893759+00:00 target192168000071 kernel: [ 1522.414658] libceph: loaded (mon/osd proto 15/24)
2018-09-28T07:28:31.895784+00:00 target192168000071 kernel: [ 1522.417950] rbd: loaded (major 252)
2018-09-28T07:28:31.909704+00:00 target192168000071 kernel: [ 1522.428345] libceph: mon0 192.168.0.69:6789 session established
2018-09-28T07:28:31.909724+00:00 target192168000071 kernel: [ 1522.430339] libceph: client14136 fsid d23ad973-6525-3200-b01c-fb09cf9c0721
2018-09-28T07:28:31.929644+00:00 target192168000071 kernel: [ 1522.450564] rbd: rbd0: capacity 1073741824 features 0x1
2018-09-28T07:28:32.157742+00:00 target192168000071 kernel: [ 1522.677784] Rounding down aligned max_sectors from 4294967295 to 4294967288
```

should we enable some debug for lrbd, systemd, kernel to see more info in logs?

**Comment by denisok** [comment]
[master.log](https://github.com/SUSE/DeepSea/files/2427755/master.log)
[kern.log](https://github.com/SUSE/DeepSea/files/2427756/kern.log)
[misc1.log](https://github.com/SUSE/DeepSea/files/2427758/misc1.log)
[minion.log](https://github.com/SUSE/DeepSea/files/2427762/minion.log)
[kern1.log](https://github.com/SUSE/DeepSea/files/2427767/kern1.log)
[misc2.log](https://github.com/SUSE/DeepSea/files/2427768/misc2.log)

**Comment by smithfarm** [comment]
@denisok I opened #1408 to get targetcli debug log messages in the teuthology.log

Also, I increased the number of test nodes (from 1 to 3) and OSDs (from 4 to 9) in the CI test cluster for both test cases that activate the igw role. So far this seems to be helping.

**Comment by smithfarm** [comment]
Fixed by a combination of lrbd 1.8-3.3.1 and deepsea 0.8.7 - specifically #1426 which removed multipath from the igw deployment test.
